File: algorithms/Minimum Depth of Binary Tree/Minimum Depth of Binary Tree.py
class Solution:
    def minDepth(self, root: Optional[TreeNode]) -> int:
        if not root:
            return 0
        
        if root.left is None:
            return self.minDepth(root.right) + 1
        if root.right is None:
            return self.minDepth(root.left) + 1
        
        l_ht = self.minDepth(root.left)
        r_ht = self.minDepth(root.right)
        
        return min(l_ht, r_ht) + 1

File: scripts/algorithms/M/Minimum Depth of Binary Tree/Minimum Depth of Binary Tree.cpp
// Runtime: 480 ms (Top 20.84%) | Memory: 144.6 MB (Top 72.25%)
class Solution {
public:
    void maxlevel(TreeNode* root,int level,int &ans){
        if(!root)
            return ;

        if(!root->left && !root->right){
            ans=min(level,ans);
            return ;
        }

        maxlevel(root->left,level+1,ans);
        maxlevel(root->right,level+1,ans);
    }
    int minDepth(TreeNode* root) {
        if(!root)
            return 0;

        int ans=INT_MAX;
        maxlevel(root,0,ans);
        return ans+1;
    }
};